packages/data-mate/src/transforms/field-transform.ts

```diff
@@ -0,0 +1,448 @@
+            return fn(data);
+        }
+
+        if (Array.isArray(data)) {
+            const results: string[] = [];
+
+            data.forEach((subData: any) => {
+                if (typeof subData === 'string') {
+                    const extractedSlice = fn(subData);
+                    if (extractedSlice) {
+                        if (Array.isArray(extractedSlice)) {
+                            results.push(...extractedSlice);
+                        } else {
+                            results.push(extractedSlice);
+                        }
+                    }
+                }
+            });
+
+            if (results.length > 0) {
+                if (isMultiValue) return results;
+                return results[0];
+            }
+        }
+
+        return null;
+    }
+
+    function matchRegex() {
+        const results = ts.matchAll(regex as string, input);
+        if (isMultiValue) return results;
+        return results ? results[0] : results;
+    }
+
+    function callExpression() {
+        try {
+            return jexl.evalSync(jexlExp as string, input);
+        } catch (err) {
+            const errMessage = `Invalid jexl expression: ${jexlExp}, error: ${err.message}`;
+            throw new ts.TSError(errMessage);
+        }
+    }
+
+    function extractAndTransferFields() {
+        let extractedResult;
+
+        if (regex) {
+            extractedResult = extractField(input, matchRegex);
+        } else if (start && end) {
+            extractedResult = extractField(input, getSubslice);
+        } else if (jexlExp) {
+            extractedResult = callExpression();
+        } else {
+            extractedResult = input;
+        }
+        return extractedResult;
+    }
+
+    const results = extractAndTransferFields();
+    if (results == null) throw new Error('Nothing to extract');
+}
+
+export function replaceRegex(input: string, {
+    regex, replace, ignoreCase, global
+}: ReplaceRegexConfig): string {
+    let options = '';
+
+    if (ignoreCase) options += 'i';
+    if (global) options += 'g';
+
+    try {
+        const re = new RegExp(regex, options);
+        return input.replace(re, replace);
+    } catch (e) {
+        throw new Error(e.message);
+    }
+}
+
+export function replaceLiteral(input: string, { search, replace }: ReplaceLiteralConfig): string {
+    try {
+        return input.replace(search, replace);
+    } catch (e) {
+        throw new Error(`Could not replace ${search} with ${replace}`);
+    }
+}
+
+export function toArray(input: string, args?: { delimiter: string }): any[] {
+    const delimiter = args ? args.delimiter : '';
+
+    return input.split(delimiter);
+}
+
+// option to specify, seconds, millisecond, microseconds?
+export function toUnixTime(input: any): number {
+    if (!isValidDate(input)) {
+        throw new Error('Not a valid date, cannot transform to unix time');
+    }
+
+    const parsed = isNaN(input) ? Date.parse(input) : Number(input);
```

Comment:
We should Number.isNaN, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN

```suggestion
    const parsed = Number.isNaN(input) ? Date.parse(input) : Number(input);
```